source/isaaclab_mimic/isaaclab_mimic/motion_planners/base_motion_planner.py

```diff
@@ -0,0 +1,130 @@
+# Copyright (c) 2024-2025, The Isaac Lab Project Developers (https://github.com/isaac-sim/IsaacLab/blob/main/CONTRIBUTORS.md).
+# All rights reserved.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import torch
+from abc import ABC, abstractmethod
+from typing import Any
+
+from isaaclab.assets import Articulation
+from isaaclab.envs.manager_based_env import ManagerBasedEnv
+
+
+class MotionPlanner(ABC):
+    """Abstract base class for motion planners.
```

Comment:
Double checking: Are there plans to add more motion planners in the future? Is this interface that has been agreed upon and does support all these future use-cases? I want to be sure, we don't define a base class that is only ever used once and only for one specific usecase. We do have a bunch of controllers/planners defined here: https://github.com/isaac-sim/IsaacLab/tree/main/source/isaaclab/isaaclab/controllers

So the need to define a new one should be motivated by future extensions :)